### ISSUE TITLE
otlpmetric: Support OTEL_EXPORTER_OTLP_PROTOCOL and OTEL_EXPORTER_OTLP_METRICS_PROTOCOL environmental variables

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-go#4641]
Reference https://github.com/open-telemetry/opentelemetry-specification/issues/3721#issuecomment-1761995814

**Comment by pellared** [comment]
I propose to add

- the protocol exporter factory that is invoked when given protocol is told to be created
- 
```go
type ProtocolExporterFactory interface {
	New(context.Conext) (trace.SpanExporter, error)
}
```

- a registry function `func RegisterProtocolExporterFactory(key string, factory ProtocolExporterFactory)` that is used to set factory for `OTEL_EXPORTER_OTLP_PROTOCOL` and `OTEL_EXPORTER_OTLP_METRICS_PROTOCOL` env var value

- an "abstract factory": `func NewFromEnv(ctx context.Context) (trace.SpanExporter, error)` that creates an OTLP exporter selected via `OTEL_EXPORTER_OTLP_PROTOCOL` or `OTEL_EXPORTER_OTLP_METRICS_PROTOCOL` env var

**Comment by pellared** [comment]
In parallel I plan to work on https://github.com/open-telemetry/opentelemetry-go/issues/4641 to have an API which would be consistent for all modules.

**Comment by pellared** [comment]
On hold. See https://github.com/open-telemetry/opentelemetry-specification/issues/3721#issuecomment-1771553380

**Comment by pellared** [comment]
Closing per https://github.com/open-telemetry/opentelemetry-specification/pull/3730
